### PR TITLE
[JEX-338] New web downloads features

### PR DESCRIPTION
**Pull request by wrightp**
This PR is aimed to add the necessary features for the new web downloads site.

- [x] product description (also added product name)
- [x] license name
- [x] license data (in metadata.json)
- [x] dependency software (in metadata.json)

I drove the changes with TDD. You'll have the review the spec changes manually since the file got lost in the PR due to all the vcr data.

Other:
- refactored ArtifactInfo class
- Fix race condition in acceptance suite

### CODE REVIEW

**Comment by wrightp** [comment]
@Padgett

File: lib/mixlib/install/artifact_info.rb [hunk 1]
```diff
@@ -19,25 +19,29 @@
...
+        version
+        license
+        url
+        md5
+        sha256
+        sha1
+        platform
+        platform_version
+        architecture
+      }.freeze
```
Comment (schisamo) [inline lib/mixlib/install/artifact_info.rb]: Can we please alphabetize this list?

File: lib/mixlib/install/backend/package_router.rb [hunk 2]
```diff
@@ -171,14 +171,17 @@ def create_artifact(artifact_map)
...
+            sha256:              artifact_map["omnibus.sha256"],
+            sha1:                artifact_map["omnibus.sha1"],
+            version:             artifact_map["omnibus.version"],
+            license:             artifact_map["omnibus.license"],
+            platform:            platform,
+            platform_version:    platform_version,
+            architecture:        normalize_architecture(artifact_map["omnibus.architecture"]),
+            url:                 chef_standard_path,
+            product_name:        options.product_name,
+            product_description: product_description
```
Comment (schisamo) [inline lib/mixlib/install/backend/package_router.rb]: Ditto on this list.

**Comment by wrightp** [comment]
@Padgett @chef/engineering-services This is ready for review

**Comment by Padgett** [comment]
This looks awesome. Thanks @wrightp! 🏆

**Comment by schisamo** [comment]
:shipit:

**Pull request merged**